policies/P.C.15.rego

```diff
@@ -0,0 +1,18 @@
+package main
+
+import data.lib.kubernetes
+
+violation[msg] {
+	kubernetes.containers[container]
+	priviledge_escalation_allowed(container)
+	msg = kubernetes.format(sprintf("%s in the %s %s allows priviledge escalation", [container.name, kubernetes.kind, kubernetes.name]))
+}
+
+priviledge_escalation_allowed(c) {
+	not kubernetes.has_field(c, "securityContext")
+}
+
+priviledge_escalation_allowed(c) {
+	kubernetes.has_field(c, "securityContext")
+	kubernetes.has_field(c.securityContext, "allowPrivilegeEscalation")
+}
```

Comment:
In rego/opa we don't have to check if intermediate keys exists in an object.
Meaning, if you want to check if ```obj.key1.key2``` you don't have to check if key1 exists in obj. You can directly check if ```obj.key1.key2 == true`` for example

Reply:
And I think a request will be violated even if it has ```allowPrivilegeEscalation = false```
So rather be specific and test
```suggestion
priviledge_escalation_allowed(c) {
  	c.securityContext. allowPrivilegeEscalation == true
  }
```